### DISCUSSION TITLE
How do I override a supported language parser?

### DISCUSSION THREAD

**Discussion by gregorias** [nvim-treesitter/nvim-treesitter#964] [Q&A] [answered]
I'd like to override the current ledger parser with [my own fork](https://github.com/gregorias/tree-sitter-ledger/tree/master). The README provides instructions on [adding a parser for unsupported language](https://github.com/nvim-treesitter/nvim-treesitter#adding-parsers), and they do not seem to work for overriding an existing parser.

I've tried

```
require'nvim-treesitter.configs'.setup {
  -- one of "all", "maintained" (parsers with maintainers), or a list of languages
  ensure_installed = "maintained",
  highlight = {
    -- false will disable the whole extension
    enable = true,
  },
}
local parser_config = require 'nvim-treesitter.parsers'.get_parser_configs()
parser_config.ledger = {
  install_info = {
    url = "https://github.com/gregorias/tree-sitter-ledger",
    files = {"src/parser.c"}
  },
  filetype = "ledger"
}
```

and reinstalling the parser with `:TSInstall ledger` but it didn't work.

**Comment by theHamsta** [comment, marked as answer]
It should work when you change the parser before the setup call. At least it work on my computer that way.
```lua
  require "nvim-treesitter.parsers".get_parser_configs().markdown = {
    install_info = {
      url = "https://github.com/ikatyang/tree-sitter-markdown",
      files = {"src/parser.c", "src/scanner.cc"}
      -- branch = "13a49d384b4ab83a5072b01e2302629c59643613"
    }
  }
  require "nvim-treesitter.configs".setup(
    {
      highlight = {
        enable = true, -- false will disable the whole extension
        disable = {} -- list of language that will be disabled
      },
     }
```

**Reply by gregorias** [reply]
I've tried putting the spec before `setup`, and that didn't work. I suspect that your example works, because "markdown" is not a parser that is in the lockfile.

**Reply by theHamsta** [reply]
It also works when I install the Julia parser from a local path or from my own fork (just tried both).

The code changing the meta table is however somehow magical, so there might be something wrong with the code dealing with the metatable.

> reinstalling the parser with :TSInstall ledger but it didn't work.

so reinstall installed the original parser, not your fork?

**Reply by gregorias** [reply]
> so reinstall installed the original parser, not your fork?

Yes. 

So I tried using a local path and it worked but switching to an https URL installs the original parser.

I've looked at the code, and see that there's some [cache involved](https://github.com/nvim-treesitter/nvim-treesitter/blob/b42d1db82d1184d832ace61fcaadc2a5ba332e0c/lua/nvim-treesitter/install.lua#L154). Could it be that that cache invalidation doesn't behave the way I would like it to?

Anyway, my immediate problem is resolved, I can just use the local path. Thank you.
